api-server-rest/src/aa.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::router::ApiHandler;
+use crate::ttrpc_proto::attestation_agent::{GetEvidenceRequest, GetTokenRequest};
+use crate::ttrpc_proto::attestation_agent_ttrpc::AttestationAgentServiceClient;
+use anyhow::*;
+use async_trait::async_trait;
+use hyper::{Body, Method, Request, Response};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+
+use crate::TTRPC_TIMEOUT;
+
+/// ROOT path for Confidential Data Hub API
+pub const AA_ROOT: &str = "/aa";
+
+/// URL for querying CDH get resource API
+const AA_TOKEN_URL: &str = "/token";
+const AA_EVIDENCE_URL: &str = "/evidence";
+
+pub struct AAClient {
+    client: AttestationAgentServiceClient,
+    accepted_method: Vec<Method>,
+}
+
+#[async_trait]
+impl ApiHandler for AAClient {
+    async fn handle_request(
+        &self,
+        remote_addr: SocketAddr,
+        url_path: &str,
+        req: Request<Body>,
+    ) -> Result<Response<Body>> {
+        if !remote_addr.ip().is_loopback() {
+            // Return 403 Forbidden response.
+            return self.forbidden();
+        }
+
+        if !self.accepted_method.iter().any(|i| i.eq(&req.method())) {
+            // Return 405 Method Not Allowed response.
+            return self.not_allowed();
+        }
+
+        let params: HashMap<String, String> = req
+            .uri()
+            .query()
+            .map(|v| form_urlencoded::parse(v.as_bytes()).into_owned().collect())
+            .unwrap_or_default();
+
+        if params.len() != 1 {
+            return self.not_allowed();
+        }
+
+        match url_path {
+            AA_TOKEN_URL => match params.get("token_type") {
+                Some(token_type) => {
+                    let results = self
+                        .get_token(token_type)
+                        .await
+                        .unwrap_or_else(|e| e.to_string().into());
+                    return self.octet_stream_response(results);
+                }
+                None => return self.bad_request(),
+            },
+            AA_EVIDENCE_URL => match params.get("runtime_data") {
+                Some(runtime_data) => {
+                    let results = self
+                        .get_evidence(&runtime_data.clone().into_bytes())
```

Comment:
This can only be exposed to the guest in particular circumstances (when the evidence is not generic). This endpoint needs to be disabled for many workloads or we will have a huge security gap. I don't see any configuration for enabling/disabling endpoints in this PR.

Reply:
Yes, agree, we plan to launch `api_server_rest` daemon only when kata agent config to enable `rest_api`, maybe we can add more config in kata agent like 
```
agent.rest_api=aa
agent.rest_api=cdh
agent.rest_api=all
```

Reply:
I added the feature flag for `api-server-rest` which allows enable different endpoints during launch time, and enable `resource` as default feature

Reply:
Ok, I am a little bit wary of adding more configs to the kernel params. I think these configs will be scoped to the runtime class, so we could potentially use a build-time configuration that is invoked when we setup the initrd, or a config file that lives in the initrd.

Reply:
Yes, agree, since kernel cmdline will impact the measurement evidence, for runtime enablement to decide whether to launch api server, finally we may need use init data(maybe part of the CoCo policy file) with integrity protection.
for build enablement, we will use a feature flag to decide whether to integrate api server binary into the rootfs.

Reply:
Ok, i think configuring via cmdline is fine as a first step. Let's make sure we add a README somewhere for how to setup all these new options and let's think about how to refine the configuration.